cli/command/container/create.go

```diff
@@ -194,31 +205,59 @@ func createContainer(ctx context.Context, dockerCli command.Cli, containerConfig
 	}
 
 	//create the container
-	response, err := dockerCli.Client().ContainerCreate(ctx, config, hostConfig, networkingConfig, opts.name)
+	var response container.ContainerCreateCreatedBody
+	if opts.pull == PullImageMissing { // Pull image only if it does not exist locally. Default.
```

Comment:
I think we can do this flow and it will clean up quite a bit:

warning, psuedo-code:

```
if PullImageAlways {
  pull()
}

err := createContainer()
if IsNotfound(err) && PullImageMissing {
  pull()
}

createContainer()
```

Reply:
Latest commit improves the flow per suggestion.

Reply:
One thing I'm not certain on is whether we should error in the case that the image is not found locally and the "never pull" option is selected? In which case the flow becomes more like so:
```
if PullImageAlways {
  pull()
}

err := createContainer()
if err {
  if IsNotfound(err) && PullImageMissing {
    pull()
    createContainer()
  }
  return err // we won't be pulling so just return whatever err we got
}
```
This also saves trying to create the container twice in the "never pull" case (ie only try to create again in the case where we might pull).